ide-backend/IdeSession/Update.hs

```diff
@@ -179,15 +179,21 @@ initSession initParams@SessionInitParams{..} ideConfig@SessionConfig{..} = do
   verifyConfig ideConfig
 
   configDirCanon <- Dir.canonicalizePath configDir
-  ideSessionDir  <- createTempDirectory configDirCanon "session."
-  let ideStaticInfo = IdeStaticInfo{..}
+  ideSessionDir' <- createTempDirectory configDirCanon "session."
+  ideSessionDir <- case configLocalWorkingDir of
```

Comment:
I don't see why the session dir needs to change in order for that to happen.  I think all that needs to happen for that is to start the ide-backend with its current directory set to the current directory, [here](https://github.com/fpco/ide-backend/pull/286/files#diff-839783ae842f246243cd9a81d9e05139R84).  So, no need to change the session directory, but this would make it so that the session directory is no longer the current 

I did a bit of code spelunking to make sure I'm not misinforming you (I didn't write this code).  Here's some of what I found:

When starting up ide-backend-server, unnamed pipes are created to communicate with it.  These are passed in as arguments: https://github.com/fpco/ide-backend/blob/bd1c780d5a4767f3e56ebd5cd35293579bea0cd2/ide-backend/IdeSession/RPC/Client.hs#L110

These file descriptors then parsed and connected to by ide-backend-server: https://github.com/fpco/ide-backend/blob/bd1c780d5a4767f3e56ebd5cd35293579bea0cd2/ide-backend-common/IdeSession/RPC/Server.hs#L45

Then, there is an initialization message which passes in the session directory: https://github.com/fpco/ide-backend/blob/bd1c780d5a4767f3e56ebd5cd35293579bea0cd2/ide-backend-server/Server.hs#L78

Here's where ide-backend-server computes the .rpc directory for communicating with a process, based on this session directory: https://github.com/fpco/ide-backend/blob/bd1c780d5a4767f3e56ebd5cd35293579bea0cd2/ide-backend-server/Server.hs#L246

So, as you can see, the session directory is entirely separate from ide-backend-session's current working directory.  In the past, it was the data directory under the session directory, but this PR will change that when the option is set.